templates/pubspecs/examples/auth_example/auth_example_flutter/pubspec.yaml

```diff
@@ -0,0 +1,105 @@
+# TEMPLATE
+
+name: auth_example_flutter
+description: A new Flutter project with Serverpod.
+
+# The following line prevents the package from being accidentally published to
+# pub.dev using `pub publish`. This is preferred for private packages.
+publish_to: 'none' # Remove this line if you wish to publish to pub.dev
+
+# The following defines the version and build number for your application.
+# A version number is three numbers separated by dots, like 1.2.43
+# followed by an optional build number separated by a +.
+# Both the version and the builder number may be overridden in flutter
+# build by specifying --build-name and --build-number, respectively.
+# In Android, build-name is used as versionName while build-number used as versionCode.
+# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
+# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
+# Read more about iOS versioning at
+# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.17.0 <4.0.0"
+  flutter: ">=3.7.0"
+
+dependencies:
+  flutter:
+    sdk: flutter
+  serverpod_flutter: VERSION
+  serverpod_auth_email_flutter: VERSION
+  serverpod_auth_shared_flutter: VERSION
+  serverpod_auth_google_flutter: VERSION
+  serverpod_auth_apple_flutter: VERSION
+  auth_example_client:
+    path: ../auth_example_client
+
+
+  # The following adds the Cupertino Icons font to your application.
+  # Use with the CupertinoIcons class for iOS style icons.
+  cupertino_icons: ^1.0.5
+
+dev_dependencies:
+  flutter_lints: ^3.0.0
+  flutter_test:
+    sdk: flutter
+
+dependency_overrides: #--CONDITIONALLY_REMOVE_LINE--#
```

Comment:
Isn't the point of the overrides that we do not need to remove them when publishing?

> If you publish a package to pub.dev, keep in mind that your package’s dependency overrides are ignored by all users of your package.

Reply:
You are correct, the only places that we really need it are the project creation files where we should always reference the published version (in production mode).

I'll slim down the removal of the dependency override to only places that are necessary.

But I'm not 100% sure what happens inside the published package. Say for example our CLI tools that is published with dependency overrides but is also bundled as a binary, I'm not sure that the binary would ignore the dependency_overrrides, so might be safer to at least strip them there.